invenio_rdm_records/contrib/journal/custom_fields.py

```diff
@@ -0,0 +1,114 @@
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Journal custom fields.
+
+Implements the following fields:
+- journal.issue
+- journal.pages
+- journal.title
+- journal.volume
+- journal.year
+"""
+
+from flask_babelex import _
+from invenio_records_resources.services.custom_fields import BaseCF
+from invenio_records_resources.services.records.facets import CFTermsFacet
+from marshmallow import fields
+from marshmallow_utils.fields import SanitizedUnicode
+
+
+class JournalCF(BaseCF):
+    """Nested custom field."""
+
+    @property
+    def field(self):
+        """Journal fields definitions."""
+        return fields.Nested(
+            {
+                "title": SanitizedUnicode(),
+                "issue": SanitizedUnicode(),
+                "volume": SanitizedUnicode(),
+                "pages": SanitizedUnicode(),
+                "year": SanitizedUnicode(),
+                "issn": SanitizedUnicode(),
+            }
+        )
+
+    @property
+    def mapping(self):
+        """Journal search mappings."""
+        return {
+            "type": "object",
+            "properties": {
+                "title": {"type": "keyword"},
+                "issue": {"type": "keyword"},
+                "pages": {"type": "keyword"},
+                "volume": {"type": "keyword"},
+                "year": {"type": "keyword"},
+                "issn": {"type": "keyword"},
+            },
+        }
+
+
+JOURNAL_NAMESPACE = {
+    # Journal
+    "journal": "",
+}
+
+
+JOURNAL_CUSTOM_FIELDS = [
+    JournalCF(name="journal:journal"),
+]
+
+JOURNAL_CUSTOM_FIELDS_UI = {
+    "section": _("Journal"),
+    "fields": [
+        {
+            "field": "journal:journal",
+            "ui_widget": "Journal",
+            "template": "journal.html",
+            "props": {
+                "label": _("Journal"),
+                "title": {
+                    "label": _("Title"),
+                    "placeholder": _("Add the journal title..."),
```

Comment:
In most other deposit form fields we don't add placeholders, and the text doesn't say anything beyond what the label says, so I wouldn't add placeholders here.